src/test/scala/is/hail/io/ExportVcfSuite.scala

```diff
@@ -174,4 +269,48 @@ class ExportVCFSuite extends SparkSuite {
       }
     }
   }
+  
+  def genFormatFieldVCF: Gen[Type] = Gen.oneOf[Type](
+      TInt32(), TFloat32(), TFloat64(), TString(), TCall(),
+      TArray(TInt32()), TArray(TFloat32()), TArray(TFloat64()), TArray(TString()), TArray(TCall()),
+      TSet(TInt32()), TSet(TFloat32()), TSet(TFloat64()), TSet(TString()), TSet(TCall()))
+  
+  def genFormatStructVCF: Gen[TStruct] =
+    Gen.buildableOf[Array, (String, Type, Map[String, String])](
+      Gen.zip(Gen.identifier,
+        genFormatFieldVCF,
+        Gen.option(
+          Gen.buildableOf2[Map, String, String](
+            Gen.zip(arbitrary[String].filter(s => !s.isEmpty), arbitrary[String])), someFraction = 0.05)
+          .map(o => o.getOrElse(Map.empty[String, String]))))
+      .filter(fields => fields.map(_._1).areDistinct())
+      .map(fields => TStruct(fields
+        .iterator
+        .zipWithIndex
+        .map { case ((k, t, m), i) => Field(k, t, i, m) }
+        .toIndexedSeq))
+  
+  @Test def testWriteGenericFormatField() {
+    val genericFormatFieldVCF:  VSMSubgen[Locus, Variant, Annotation] = VSMSubgen.random.copy(
+      tSigGen = genFormatStructVCF,
+      tGen = (t: Type, v: Variant) => t.genValue.resize(20))
+    
+    val untestedFields = Set("rsid", "qual", "info")
+    
+    val out = tmpDir.createTempFile("foo", "vcf.bgz")
+    val p = forAll(VariantSampleMatrix.gen(hc, genericFormatFieldVCF), Gen.choose(1, 10),
+      Gen.choose(1, 10)) { case (vkds, nPar1, nPar2) =>
+      println(vkds.rowType)
+      
+      vkds.vaSignature match {
+        case t: TStruct if untestedFields.intersect(t.fieldNames.toSet).isEmpty => 
```

Comment:
does this only run the test if VSMSubgen.random manages to generate at least one identifier in `untestedFields`?

Reply:
Ah, good catch! I meant nonEmpty. But this is silly, I've now changed it to just enforce that va is empty.